src/main/java/com/cascadebot/cascadebot/commandmeta/Argument.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package com.cascadebot.cascadebot.commandmeta;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class Argument {
+
+    private final String arg;
+    private final String description;
+    private final Set<Argument> subArgs;
+    private final ArgumentType type;
+    private final Set<String> aliases;
+
+    private Argument(String arg, String description, Set<Argument> subArgs, ArgumentType type, Set<String> aliases) {
+        this.arg = arg.toLowerCase(); //This probably isn't needed but megh
+        this.description = description;
+        this.subArgs = Collections.unmodifiableSet(subArgs);
+        if (subArgs.size() > 0) {
+            this.type = ArgumentType.COMMAND;
+        } else {
+            this.type = type;
+        }
+        this.aliases = Collections.unmodifiableSet(aliases);
+    }
+
+    public static Argument of(String arg, String description) {
+        return new Argument(arg, description, Set.of(), ArgumentType.COMMAND, Set.of());
+    }
+
+    public static Argument ofA(String arg, String description, Set<String> aliases) {
+        return new Argument(arg, description, Set.of(), ArgumentType.COMMAND, aliases);
+    }
+
+    public static Argument of(String arg, String description, ArgumentType type) {
+        return new Argument(arg, description, Set.of(), type, Set.of());
+    }
+
+    public static Argument ofA(String arg, String description, ArgumentType type, Set<String> aliases) {
+        return new Argument(arg, description, Set.of(), type, aliases);
+    }
+
+    public static Argument of(String arg, String description, Set<Argument> subArgs) {
+        return new Argument(arg, description, subArgs, ArgumentType.COMMAND, Set.of());
+    }
+
+    public static Argument ofA(String arg, String description, Set<Argument> subArgs, Set<String> aliases) {
+        return new Argument(arg, description, subArgs, ArgumentType.COMMAND, aliases);
+    }
+
+    /**
+     * Gets the usage string.
```

Comment:
Could you give an example of how each argument is represented as well as an example showing how a line would be constructed?

Reply:
Not really without having a 50 line documentation as it's really different depending on what the args is if it has sub args if it has a description. if you can come up with a way to do so feel free to put it in here.